main.cpp

```diff
@@ -92,6 +92,91 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    FloatType() {}
+
+    float add(float lhs, float rhs)
+    {
+        return lhs + rhs;
+    }
+
+    float subtract(float lhs, float rhs)
+    {
+        return lhs - rhs;
+    }
+
+    float multiply(float lhs, float rhs)
+    {
+        return lhs * rhs;
+    }
+
+    float divide(float lhs, float rhs)
+    {
+        if(rhs == 0.0f) std::cout << "\nwarning, floating point division by zero returns 'inf' !\n";
+        return lhs / rhs;
+    }
+};
+
+struct IntType
+{
+    IntType() {}
+
+    int add(int lhs, int rhs)
+    {
+        return lhs + rhs;
+    }
+
+    int subtract(int lhs, int rhs)
+    {
+        return lhs - rhs;
+    }
+
+    int multiply(int lhs, int rhs)
+    {
+        return lhs * rhs;
+    }
+
+    int divide(int lhs, int rhs)
+    {
+        if(rhs == 0) 
+        {
+            std::cout << "error, integer division by zero will crash the program!\n";
+            std::cout << "returning lhs\n";
+            return lhs;
+        }
+        else return lhs / rhs;
+    }
+};
+
+struct DoubleType
+{
+DoubleType() {}
+
+double add(double lhs, double rhs)
+    {
+        return lhs + rhs;
+    }
+
+    double subtract(double lhs, double rhs)
+    {
+        return lhs - rhs;
+    }
+
+    double multiply(double lhs, double rhs)
+    {
+        return lhs * rhs;
+    }
+
+    double divide(double lhs, double rhs)
+    {
+        if(rhs == 0.0) std::cout << "\nwarning, floating point division by zero returns 'inf' !\n";
+        return lhs / rhs;
+    }
```

Comment:
put single-line if() blocks on their own line
This makes the code more readable and logic easier to understand
- [ ] revise